sdk/identity/azure-identity/src/main/java/com/azure/identity/ManagedIdentityCredential.java

```diff
@@ -36,14 +36,19 @@
             .identityClientOptions(identityClientOptions)
             .build();
         Configuration configuration = Configuration.getGlobalConfiguration().clone();
-        if (configuration.contains(Configuration.PROPERTY_MSI_ENDPOINT)
-                || (configuration.contains(Configuration.PROPERTY_IDENTITY_ENDPOINT)
-                        && configuration.contains(Configuration.PROPERTY_IDENTITY_HEADER))) {
-            appServiceMSICredential = new AppServiceMsiCredential(clientId, identityClient);
-            virtualMachineMSICredential = null;
+        if (configuration.contains(Configuration.PROPERTY_IDENTITY_ENDPOINT)) {
+            if (configuration.contains(Configuration.PROPERTY_IDENTITY_HEADER)) {
+                managedIdentityServiceCredential = new AppServiceMsiCredential(clientId, identityClient);
+            } else if (configuration.contains(PROPERTY_IMDS_ENDPOINT)) {
+                managedIdentityServiceCredential = new ArcIdentityCredential(clientId, identityClient);
+            } else {
+                managedIdentityServiceCredential = null;
```

Comment:
If `managedIdentityServiceCredential` is null, getToken() method throws an exception. So, why not throw that exception here itself? Why delay throwing exception until later? Since this is `final` it won't change after the constructor is called anyway.

Reply:
yeah, this is by design.
We don't throw exceptions beyond user input validation at build/construction level.
DefaultAzureCredential will start to break at the construction level too, which shouldn't happen and this will be a breaking change as well.